### ISSUE TITLE
Updated Autocomplete and Filters to Better Support Screen readers

### ISSUE THREAD

**Issue by bbertucc** [bbertucc/equalify#398]
As noted in #374, autocomplete causes confusion with screen reader users. To fix that confusion, I suggest the following:

- [x] Add a "Selected Filters" heading in a "selected filters" landmark region when you get to the page with a list of selected filters 
- [x] Add description text to the filter autocomplete field that says "Search and browse filters then add multiple filters to reports."
- [x] When you select a filter, announce to screen reader users: "Filter Name was added to Selected Filters. Type to search for additional filters. Use the arrow keys to select from results. Save your updates by clicking the Update Report button."

**Comment by wilsuriel03** [comment]
working on it

**Comment by wilsuriel03** [comment]
complete ✅

**Comment by bbertucc** [comment]
Great. Switching to @kevinandrews1 to make sure he can use the autocomplete.

**Comment by kevinandrews1** [comment]
@wilsuriel03 This is fine but the text search and browse filters... should be associated with the autocomplete using aria-describedby. It provides a semantic relationship between the input and the on-screen text, which is read after the input and how to interact with it (see the [video](https://drive.google.com/file/d/1WUSJMe9BE8e29V0cA3uqmkwKfG18COpJ/view?usp=drive_link) to help clarify), behaving as helper text. The issue mainly comes when I tab off the autocomplete and return to it. It's collapsed and no matter what I do with the keyboard I can't reexpose the list of items--for example the messages.

**Comment by wilsuriel03** [comment]
@kevinandrews1 

- I've associated the input with descriptive text using the aria-describedby attribute. 

- The suggestion box now appears when the input gains focus, giving users immediate visibility of available filters without needing to type first.

- I've ensured that the suggestion box remains accessible and navigable when the input is focused, allowing for better keyboard navigation and interaction.


These changes aim to provide a more intuitive and accessible experience. Please let me know if there are any further improvements I can make. For the upcoming release, we should aim to include only essential changes to ensure timely delivery. Any additional improvements or smaller tweaks can be addressed post-release as part of our maintenance cycle.

**Comment by kevinandrews1** [comment]
Hey @wilsuriel03, thanks. I agree with you we need to stick to the timeline. There's inconsistent behavior with respect to aria-describedby and its support. I'd need some time to research and to play with this and we're out of time at the moment - for reference, it may (or may not) be related to this [issue](https://github.com/FreedomScientific/standards-support/issues/839) on the Freedom Scientific GitHub. I think this is fine to get us through the initial v1 release.

## JAWS speech output:
> Select Filter Type Combo box
> Messages
> To change the selection use the Arrow keys.
> Type to search edit combo
> collapsed
blank
> To set the value use the Arrow keys or type the value.

## NVDA speech output:
> Select Filter Type  combo box  Messages  collapsed  opens list
> Type to search  combo box  collapsed  has auto complete  editable  Search and browse filters then add multiple filters to reports.  blank

## Additional Notes
As you can see, JAWS does not speak the content associated via aria-describedby whereas NVDA does. Maybe it's to do with it being on a combobox control or the role of combobox, but this would require additional research and testing on my end. I think we should revisit the autocomplete component overall. If you watch the [video](https://drive.google.com/file/d/1WUSJMe9BE8e29V0cA3uqmkwKfG18COpJ/view?usp=drive_link) I provided previously, pressing the remove filter button does communicate the removal but it also seems to read off all the other selected filters on the screen. All the user needs to know is that "filter name" was removed. I'm closing this in favor of plowing ahead, but issues that need to be created. Pinging @bbertucc for visibility.
- When the combobox receives focus, a user tabs away from it, and returns focus to the combobox, they can no longer interact with the items in the list.
- Why is the focus outline set to none? Removing the focus outline on input elements is bad for accessibility because it makes it difficult or impossible for keyboard and screen reader users to navigate forms, as they rely on visible focus indicators to understand which element is currently active.
- Verbose feedback when pressing remove filters.

**Comment by alexstine** [comment]
@kevinandrews1 I came to this late but you are correct in your statement. Every screen reader, based on verbosity settings, handles `aria-describedby` attribute differently. Voiceover for example defaults to speaking that the item has a help tag and then it is your choice if you wish to hear the description or not. NVDA probably has the best out of box support for this attribute. As far as any usage of `aria-description`, it should be avoided as it lacks substantial support across assistive tech.

Also worth pointing out, I don't know how the code looks today, but you cannot conditionally render `aria-describedby`. It should always be available in the DOM for best support. If there is an input field involved, my guess is `role="combobox"` might have not been the best role choice to use. Building accessible complex auto completes is very tricky and requires a ton of debugging/development hours. If this is still an issue, this is something I'm happy to put in my backlog to address.

CC: @bbertucc

**Comment by bbertucc** [comment]
All good notes to remember. Thanks @alexstine!
